### DISCUSSION TITLE
Localization issue trying to filter by AutoRefinableString in search box with values in other language

### DISCUSSION THREAD

**Discussion by Juanencm** [microsoft-search/pnp-modern-search#4120] [Q&A]
__Remove the above section before submitting, and removal of the below template will result in the issue being closed.__

### Follow the template below or we will have to delete this request!!
We can assist a lot faster if you provide the following information:

**Version used**
Ex: 4.14.0

**Describe the bug**
- We have 5 site collections in SharePoint Online grouped in a hub
- The sites have been configured multilanguage in Spanish (default) and English
- All the libraries and lists in these sites have a column of type metadata to categorize the contents with multiple tags, the column internal name is WAVE_KEYWORDS and is attached to a Global term group in the Term store
- We have mapped the following properties in the managed properties of search schema at tenant level:
RefinableString01 --> OWS_TAXID_WAVE_KEYWORDS
- We have created two search pages, one for Spanish and another for English, with searh box, search results and search refiners webparts
- In the searh results webpart we have selected the fields/properties RefinableString01 to be used in the template
- We have modified the template of the search result webpart, in order to show the associated keywords in the data of each result. Each keyword its formatted as a link, because we want to search by the keyword when you click on it. We are using the following code for that, using the "Auto" prefix to show the OWS_TAXID_WAVE_KEYWORDS values in English
<div>
	{{#each (split (slot item @root.slots.AutoRefinableString01) "ǂǂ") as |Autokeywords| }}
		<span>
			<a href="?q=RefinableString01:%22{{trim Autokeywords}}%22&vertical=All" class="template--listItem--tag" rel="noopener noreferrer">
				{{trim Autokeywords}}
			</a>
		</span>
	{{/each}}
</div>
- The results are showing ok, the tags data are showing ok in each result, but in the English version, when we click on the custom generated link for each tag, the associated search is not working, due to it seems it only works if we use the value of the tag in Spanish in the value that is inserted in the search box. For example, using in the search box the value RefinableString01:"Talent" is not working, we have to use RefinableString01:"Talento" (the value of the tag in spanish)
- We are testing with a user with his regional and language configured to English
- We have re-indexed the libraries and lists in SharePoint multiple times

**To Reproduce**
1. Configure a Global Search term with values in Spanish (default) and English
2. Use a column of metadata type associated to the search term
3. Configure and associate to RefinableSting in search managed properties the ows_ and ows_taxid properties
4. Create a search page and use the RefinableStrings in the searh results webpart
6. Edit the template to show for each results the Keywords field values with a link in order to be able of filtering by the clicked tag/value using the AutoRefinableString01 field/property

**Expected behavior**
When we click on the link of each tag, the results must be updated filtering by the tag/valued clicked, but in English version it does not work, it only works if we mannually change the generated value by the link in the search box to use the tag value in spanish

**Screenshots**
Search schema configuration:
![image](https://github.com/user-attachments/assets/f1427b26-072a-4b6c-a8ea-c5956e993d1e)

Properties selected in the searh results web part:
![image](https://github.com/user-attachments/assets/a181fd59-14ec-482f-a639-b18159696b7c)

Template modified with the described code above:
![image](https://github.com/user-attachments/assets/5f5eab6d-da05-4305-81b7-bb1e0877c2d3)

Screenshot of the initial search results in English:
![image](https://github.com/user-attachments/assets/49e9f96a-4194-4b29-99a4-ca314854f304)

Screenshot when we have clicked on the link of tag "Talent":
![image](https://github.com/user-attachments/assets/1b51dc07-2ca6-44ea-bcf9-1c46ecfb7aaa)

Screenshot if we change te value to use the value of the tem in spanish:
![image](https://github.com/user-attachments/assets/8ff07b19-2d1b-44d0-8a7f-658e0c31f66b)

**Desktop (please complete the following information):**
 - Browsers used: Edge, Firefox, Chrome

**Additional context**

**Comment by wobba** [comment]
Label names are indexed in the default language the site was created in if a label exists in this language. For multi-language you should use the taxid as the filter/query to match on terms used independent of the language. Did you enable localization in the result webpart? (https://microsoft-search.github.io/pnp-modern-search/usage/search-results/data-sources/sharepoint-search/)

When clicking a label this means the ideal query is the term guid, not the label.

**Reply by Juanencm** [reply]
Hello, thank you for your quick response, answering your question, yes, we have enabled the localization option in the search results webpart, because if its disabled the AutoRefinableString01 property is empty.

Based on your response, we should use the termguid to generate the link, our problem is that we do not see how to do that:

- If we use the RefinableString01 (OWS_TAXID_WAVE_KEYWORDS) to get the field values, we get a collection of data with the labels in Spanish with the following format:

RefinableString01 (ows-taxid-wave-keywords):
GP0|#56fd82a8-d26d-495f-a5bd-77fb9a2431d5
L0|#056fd82a8-d26d-495f-a5bd-77fb9a2431d5|LabelSpanish1
GTSet|#d40d5b5d-6f4c-4fb3-a6b7-ec6ac3c59163
GP0|#6408f02d-f521-4b3b-a4b1-aca3d29d6704
L0|#06408f02d-f521-4b3b-a4b1-aca3d29d6704|LabelSpanish2
GP0|#9ffe833c-97b6-4648-98b6-6b1720dba08f
L0|#09ffe833c-97b6-4648-98b6-6b1720dba08f|LabelSpanish3

- So in order to show the values in English we have to use the AutoRefinableString01 property, but it seems this property only contains the collection of labels in English, without the TermGuid, so we cannot generate a link using the TermGuid and filter the data by the keyword:

AutoRefinableString01:
LabelEnglish1ǂǂ LabelEnglish2ǂǂ LabelEnglish3

**Reply by wobba** [reply]
The fix would be that instead of the "label" when clicking it should use the guid, so the query would end up `RefinableString01:<guid>` which would not look good from a UX perspective, but would work. I believe changing the display template somewhat could make this work. @kasperbolarsen, as you have looked more at this code lately, what do you think?

**Reply by Juanencm** [reply]
Hi wobba, sorry but I don't know how we can use the guid instead the label if we only have the AutoRefinableString01 property to show the values in english, due to this property values does not contain the associated guid of each label

**Reply by wobba** [reply]
I was thinking to manually change the handlebar template to pull out the id and use it. I haven't looked into it myself, but that's where I would start looking.

**Reply by Juanencm** [reply]
Hi wobba, I am no sure if I understand you with "the handlebar template", we have been trying in the search result template to search in the RefinableString01 by the order of the AutoRefinableString01 selected element in order to get the TermSet, but it seemns that the order of the elements contained in the AutoRefinableString01 (with the english values) can change and is not the same of the RefinableString01 elements, so we cannot match them to get the Id.

What we are trying to do it was possible in previous version 3, and in this new version with the use of the "Auto" property containing only the labels I dont see that it will be possible, thats why I initially opened as an issue.

**Reply by wobba** [reply]
@kasperbolarsen do you have an idea on how clicking a tag on a result can use the taxid or original language label (spanish in this case) when the label is clicked?

**Reply by kasperbolarsen** [reply]
I'll look into it
